src/Symfony/Component/Security/Core/Authentication/AuthenticationTrustResolver.php

```diff
@@ -23,10 +25,18 @@ class AuthenticationTrustResolver implements AuthenticationTrustResolverInterfac
     private $anonymousClass;
     private $rememberMeClass;
 
-    public function __construct(string $anonymousClass, string $rememberMeClass)
+    public function __construct(string $anonymousClass = null, string $rememberMeClass = null)
     {
         $this->anonymousClass = $anonymousClass;
         $this->rememberMeClass = $rememberMeClass;
+
+        if (null !== $anonymousClass && !is_a($anonymousClass, AnonymousToken::class, true)) {
+            @trigger_error(sprintf('Configuring a custom anonymous token class is deprecated since Symfony 4.2. To fix this, have the %s class extend the %s class instead, and remove the %s constructor argument.', $anonymousClass, AnonymousToken::class, self::class), E_USER_DEPRECATED);
```

Comment:
We try to have only one sentence for deprecations: `Configuring a custom anonymous token class is deprecated since Symfony 4.2; have the "%s" class extend the "%s" class instead, and remove the "%s" constructor argument.` (note the quotes around class names and parameters).

Reply:
That makes sense, I'll update it :+1:

Reply:
Noticed that the first argument of the sprintf was the wrong one of for the remember me as well, it received the `$anonymousClass` instead (fixed)